view/next-project/src/pages/year_periods/index.tsx

```diff
@@ -0,0 +1,181 @@
+import clsx from 'clsx';
+import Head from 'next/head';
+import { useRouter } from 'next/router';
+import { useEffect, useState, useMemo } from 'react';
+import { useRecoilValue } from 'recoil';
+
+import OpenAddModalButton from '@/components/year_periods/OpenAddModalButton';
+import OpenDeleteModalButton from '@/components/year_periods/OpenDeleteModalButton';
+import OpenEditModalButton from '@/components/year_periods/OpenEditModalButton';
+import { authAtom } from '@/store/atoms';
+import { getCurrentUser } from '@/utils/api/currentUser';
+import { get } from '@api/api_methods';
+import { Card, Title } from '@components/common';
+import MainLayout from '@components/layout/MainLayout/MainLayout';
+import { YearPeriods, User } from '@type/common';
+
+interface Props {
+  yearPeriods: YearPeriods[];
+}
+
+export const getServerSideProps = async () => {
+  const getPeriodURL = process.env.SSR_API_URI + '/years/periods';
+  const periodsRes = await get(getPeriodURL);
+
+  return {
+    props: {
+      yearPeriods: periodsRes,
+    },
+  };
+};
+
+export default function Periods(props: Props) {
+  const { yearPeriods } = props;
+  const router = useRouter();
+
+  const auth = useRecoilValue(authAtom);
+  const [currentUser, setCurrentUser] = useState<User>();
+
+  const formatYearPeriods =
+    yearPeriods &&
+    yearPeriods.map((yearPeriod) => {
+      return {
+        ...yearPeriod,
+        startedAt: yearPeriod.startedAt && new Date(yearPeriod.startedAt).toLocaleDateString('ja'),
+        endedAt: yearPeriod.endedAt && new Date(yearPeriod.endedAt).toLocaleDateString('ja'),
```

Comment:
この書き方だと`yearPeriod.startedAt`がundefined(falsy)な場合はstartedAt=undefinedとなってしまいますが想定した実装ですか？
```suggestion
        startedAt: yearPeriod.startedAt || new Date(yearPeriod.startedAt).toLocaleDateString('ja'),
        endedAt: yearPeriod.endedAt || new Date(yearPeriod.endedAt).toLocaleDateString('ja'),
```
ではありませんか？

Reply:
修正しました [[fix]type名変更](https://github.com/NUTFes/FinanSu/pull/660/commits/0668b50cd8e3e3ab0a9718d1adb9bcbb8d3be4bd) [[fix]レコードが0の場合の処理追加](https://github.com/NUTFes/FinanSu/pull/660/commits/2f29a6c71b8aa27d174ccac85925bca655872157)
yearPeriodの型がこれまでだとyearPeriod.startedAtがundefinedの可能性があったため上のようにしていました。

Reply:
> yearPeriodの型がこれまでだとyearPeriod.startedAtがundefinedの可能性があったため上のようにしていました

なるほど、ずっと「startedAtが”ない場合”は仮のDateをいれる」かと思ってたんですが、「startedAtが"ある場合"は日本語の時間にする」みたいな処理だったんですね、誤読でした